joblib/parallel.py

```diff
@@ -249,6 +252,13 @@ def __call__(self):
             return [func(*args, **kwargs)
                     for func, args, kwargs in self.items]
 
+    def __reduce__(self):
+        if self._reducer_callback is not None:
+            self._reducer_callback()
+            # No need to pickle the callback
+            self._reducer_callback = None
```

Comment:
Good catch, hackish but good enough.

Reply:
This means that it will work only if it is pickled a single time for the loky backend.
Why not simply return `BatchedCalls, (iterator_slice, backend_and_jobs, None pickle_cache)`. This would do the same without over-writting the callback no?

Reply:
> Why not simply return BatchedCalls, (iterator_slice, backend_and_jobs, None pickle_cache).

Sure, we can also do that. 

> Eventhough I agree it will probably be pickled only once for the loky backend in practice. But for other backend such as dask, I am not sure.

Note that for now, this `callback` is set to `None` for all backends other than `loky`.